service/javadsl/kafka/server/src/test/scala/com/lightbend/lagom/internal/javadsl/broker/kafka/JavadslKafkaApiSpec.scala

```diff
@@ -243,12 +246,12 @@ class JavadslKafkaApiSpec
       initialOffset shouldBe NoOffset
 
       // Put some messages in the stream
-      test3EventJournal.append("firstMessage")
-      test3EventJournal.append("secondMessage")
-      test3EventJournal.append("thirdMessage")
+      test3EventJournal.append(("firstMessage", true))
+      test3EventJournal.append(("secondMessage", true))
+      test3EventJournal.append(("thirdMessage", false))
 
       // Wait for a subscriber to consume them all (which ensures they've all been published)
-      val allMessagesReceived = new CountDownLatch(3)
+      val allMessagesReceived = new CountDownLatch(2)
```

Comment:
Because the last message is not emited, this countdown latch may open too early.  The solution is the `eventually` you used below. 

Alternatively, there could be a `(("fourthmessage", true))` so the `CountDownLatch(3)` is still awaiting until `3`.

Reply:
Changed to:
```
test3EventJournal.append(("firstMessage", true))
test3EventJournal.append(("secondMessage", false))
test3EventJournal.append(("thirdMessage", true))
```